dds/DCPS/XTypes/TypeLookupService.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+#ifndef OPENDDS_DCPS_TYPE_LOOKUP_SERVICE_H
+#define OPENDDS_DCPS_TYPE_LOOKUP_SERVICE_H
+
+/*
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#include "TypeObject.h"
+
+#include <dds/DCPS/RcObject.h>
+#include <dds/DCPS/SequenceNumber.h>
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace XTypes {
+
+class OpenDDS_Dcps_Export TypeLookupService : public virtual DCPS::RcObject {
+public:
+  TypeLookupService();
+  ~TypeLookupService();
+
+  // For TypeAssignability
+  const TypeObject& GetTypeObject(const TypeIdentifier& type_id) const;
+  void AddTypeObjectsToCache(const TypeIdentifier& ti, const TypeObject& tobj);
```

Comment:
Does the spec specify that the functions have to be capitalized? If not could you make them either snake or lowercase camel case.